OptimizelySDK/ProjectConfig.cs

```diff
@@ -353,5 +362,114 @@ public Variation GetVariationFromId(string experimentKey, string variationId)
+        /// <param name="userId">The user ID</param>
+        /// <returns>Variation entity which the given user and experiment should be forced into.</returns>
+        public Variation GetForcedVariation(string experimentKey, string userId)
+        {
+            if (string.IsNullOrEmpty(userId))
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format("User ID is invalid.", userId));
+                return null;
+            }
+
+            if (_ForcedVariationMap.ContainsKey(userId) == false)
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"User ""{0}"" is not in the forced variation map.", userId));
+                return null;
+            }
+
+            Dictionary<string, string> experimentToVariationMap = _ForcedVariationMap[userId];
+
+            string experimentId = GetExperimentFromKey(experimentKey).Id;
+
+            // this case is logged in getExperimentFromKey
+            if (string.IsNullOrEmpty(experimentId))
+                return null;
+            
+            if (experimentToVariationMap.ContainsKey(experimentId) == false)
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"No experiment ""{0}"" mapped to user ""{1}"" in the forced variation map.", experimentKey, userId));
+                return null;
+            }
+            
+            string variationId = experimentToVariationMap[experimentId];
+
+            if (string.IsNullOrEmpty(variationId))
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"No variation mapped to experiment ""{0}"" in the forced variation map.", experimentKey));
+                return null;
+            }
+            
+            string variationKey = GetVariationFromId(experimentKey, variationId).Key;
+
+            // this case is logged in getVariationFromKey
+            if (string.IsNullOrEmpty(variationKey))
+                return null;
+
+            Logger.Log(LogLevel.DEBUG, string.Format(@"Variation ""{0}"" is mapped to experiment ""{1}"" and user ""{2}"" in the forced variation map", variationKey, experimentKey, userId));
+            
+            Variation variation = GetVariationFromKey(experimentKey, variationKey);
+
+            return variation;
+        }
+
+        /// <summary>
+        /// Sets an associative array of user IDs to an associative array of experiments to forced variations.
+        /// </summary>
+        /// <param name="experimentKey">The experiment key</param>
+        /// <param name="userId">The user ID</param>
+        /// <param name="variationKey">The variation key</param>
+        /// <returns>A boolean value that indicates if the set completed successfully.</returns>
+        public bool SetForcedVariation(string experimentKey, string userId, string variationKey)
+        {
+            if (string.IsNullOrEmpty(userId))
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format("User ID is invalid.", userId));
```

Comment:
Remove string format as `userId` is not being used.

Reply:
Done.